tests/Client/UpdaterTest.php

```diff
@@ -446,33 +702,160 @@ public function providerDelegationErrors(): array
+                'f.txt',
+                [
+                    'root' => 2,
+                    'timestamp' => 2,
+                    'snapshot' => 2,
+                    'targets' => 2,
+                    'a' => 1,
+                    'b' => 1,
+                    'c' => 1,
+                    'd' => 1,
+                    'e' => null,
+                    'f' => null,
+                ],
+            ],
+            // Test cases using the 'TUFTestFixtureTopLevelTerminating' fixture set.
+            'TUFTestFixtureTopLevelTerminating b.txt' => [
+                'TUFTestFixtureTopLevelTerminating',
+                'b.txt',
+                [
+                    'root' => 2,
+                    'timestamp' => 2,
+                    'snapshot' => 2,
+                    'targets' => 2,
+                    'a' => 1,
+                    'b' => null,
+                ],
+            ],
+            // Test cases using the 'TUFTestFixtureNestedTerminatingNonDelegatingDelegation' fixture set.
+            'TUFTestFixtureNestedTerminatingNonDelegatingDelegation c.txt' => [
+                'TUFTestFixtureNestedTerminatingNonDelegatingDelegation',
+                'c.txt',
+                [
+                    'root' => 2,
+                    'timestamp' => 2,
+                    'snapshot' => 2,
+                    'targets' => 2,
+                    'a' => 1,
+                    'b' => 1,
+                    'c' => null,
+                    'd' => null,
+                ],
+            ],
+            'TUFTestFixtureNestedTerminatingNonDelegatingDelegation d.txt' => [
+                'TUFTestFixtureNestedTerminatingNonDelegatingDelegation',
+                'd.txt',
+                [
+                    'root' => 2,
+                    'timestamp' => 2,
+                    'snapshot' => 2,
+                    'targets' => 2,
+                    'a' => 1,
+                    'b' => 1,
+                    'c' => null,
+                    'd' => null,
+                ],
+            ],
+            // Test cases using the 'TUFTestFixture3LevelDelegation' fixture set.
+            // A search for non existent target should that matches the paths
+            // should search the complete tree.
+            'TUFTestFixture3LevelDelegation z.txt' => [
+                'TUFTestFixture3LevelDelegation',
+                'z.txt',
+                [
+                    'root' => 2,
+                    'timestamp' => 2,
+                    'snapshot' => 2,
+                    'targets' => 2,
+                    'a' => 1,
+                    'b' => 1,
+                    'c' => 1,
+                    'd' => 1,
+                    'e' => 1,
+                    'f' => 1,
+                ],
+            ],
+            // A search for non existent target that does match the paths
+            // should not search any of the tree.
+            'TUFTestFixture3LevelDelegation z.zip' => [
```

Comment:
I think that we should change these keys to actually describe the expected outcome. For example, this one could maybe be "non-existent target, does not match paths, tree not searched" or something like that. Just naming these data sets after the fixture and target name, is simply repeating arguments that are already in the data set.

Reply:
I'd be okay doing this in a follow-up.